src/mac/frame.rs

```diff
@@ -346,6 +346,10 @@ impl Header {
             .ok_or(DecodeError::InvalidFrameVersion(frame_version))?;
         len += 2;
 
+        if buf.len() < 3 {
+            return Err(DecodeError::NotEnoughBytes);
+        }
+
```

Comment:
Can't we update the check at the top at the method instead? Or is it important we see other decoding errors that could happen first?